src/ruby/PreProcessor.rb

```diff
@@ -25,7 +26,7 @@ def self.reset
   # Process the emerald to remove indentation and replace with brace convention
   # for an easier time parsing with context free grammar
   def self.process_emerald(file_name)
-    input = File.open(file_name, 'r').read
+    input = File.open(file_name, 'r').read.gsub('{', '\{')
```

Comment:
Out of curiosity does this actually work yet? Like does the grammar look for `{` and also make sure there isn't a backslash preceding it?